src/base.jl

```diff
@@ -33,6 +33,8 @@ ncclRedOp_t(::typeof(+)) = ncclSum
 ncclRedOp_t(::typeof(*)) = ncclProd
 ncclRedOp_t(::typeof(max)) = ncclMax
 ncclRedOp_t(::typeof(min)) = ncclMin
+# Handles the case where user directly passed in the ncclRedOp_t (eg. `NCCL.avg`)
+ncclRedOp_t(x::ncclRedOp_t) = x
```

Comment:
This also shouldn't be required when the conversion happens implicitly using `convert`. But since this doesn't change the end-user API, we can change that later.

Reply:
https://github.com/JuliaGPU/NCCL.jl/blob/3a49978813c5d77a45b77a29d47a4f751b1aec78/src/collective.jl#L17 was causing trouble for me when I passed in NCCL.avg

Reply:
Yeah I know, I'm just saying that there's a better fix than adding a no-op function, namely by relying on `convert` to do the conversion when necessary. It's fine adding this now, it was more of a mental note that we need to rework the conversion at some later point :-)